src/main/java/com/dabsquared/gitlabjenkins/trigger/handler/push/PushHookTriggerHandlerFactory.java

```diff
@@ -12,20 +13,20 @@
 
     private PushHookTriggerHandlerFactory() {}
 
-    public static PushHookTriggerHandler newPushHookTriggerHandler(boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
+    public static PushHookTriggerHandler newPushHookTriggerHandler(GitLabPluginMode gitLabPluginMode, boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
         if (triggerOnPush || triggerOpenMergeRequestOnPush == TriggerOpenMergeRequest.both) {
-            return new PushHookTriggerHandlerList(retrieveHandlers(triggerOnPush, triggerOpenMergeRequestOnPush));
+            return new PushHookTriggerHandlerList(retrieveHandlers(gitLabPluginMode, triggerOnPush, triggerOpenMergeRequestOnPush));
         } else {
             return new NopPushHookTriggerHandler();
         }
     }
 
-    private static List<PushHookTriggerHandler> retrieveHandlers(boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
+    private static List<PushHookTriggerHandler> retrieveHandlers(GitLabPluginMode gitLabPluginMode, boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
         List<PushHookTriggerHandler> result = new ArrayList<PushHookTriggerHandler>();
         if (triggerOnPush) {
             result.add(new PushHookTriggerHandlerImpl());
         }
-        if (triggerOpenMergeRequestOnPush == TriggerOpenMergeRequest.both) {
+        if (gitLabPluginMode == GitLabPluginMode.LEGACY && triggerOpenMergeRequestOnPush == TriggerOpenMergeRequest.both) {
```

Comment:
You are right. The case I was referring was new commits on the source branch (the user's fork). That gets handled by the MR trigger. I'll take a look at this, it might be a limitation of the "Modern" approach (if that's the case, I'll document it to reflect it).